### PR TITLE
[docs] nn.functional.embedding: Note expected discrepancy between numerical and analytical gradients 

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #99745
* __->__ #99181
* 

Fixes https://github.com/pytorch/pytorch/issues/93950

### CODE REVIEW

File: torch/nn/functional.py [hunk 1]
```diff
@@ -2138,6 +2138,11 @@ def embedding(
 
     See :class:`torch.nn.Embedding` for more details.
 
+    .. warning::
+        To ensure correct gradients during backward, make sure the entries in the :attr:`weight`
+        matrix at the row specified by :attr:`padding_idx` are all zeroes. Alternatively, use
```
Comment (albanD) [inline torch/nn/functional.py:2143]: Is that full rows or any arbitrary subset of the input?

Comment (soulitzer) [reply torch/nn/functional.py]: Full rows, the embedding matrix is 2D, and padding_idx gives us the row

**Comment by soulitzer** [comment]
See updated commentn in https://github.com/pytorch/pytorch/issues/93950#issuecomment-1454227165

**Review by albanD — APPROVED** [review]
SGTM!
